src/lib/util/talloc.c

```diff
@@ -723,9 +723,17 @@ void **talloc_array_null_terminate(void **array)
 	len = talloc_array_length(array);
 	ctx = talloc_parent(array);
 
-	new = talloc_realloc_fn(ctx, array, len + 1);
+	new = talloc_array(ctx, void *, len + 1);
```

Comment:
Can you try this with `void *_talloc_realloc_array(const void *ctx, void *ptr, size_t el_size, unsigned count, talloc_get_name(ptr));`